src/main/java/in/projecteka/consentmanager/user/PatientsController.java

```diff
@@ -59,22 +72,104 @@
+            return Mono.empty();
+        }
+
+        return Mono.just(GenerateOtpResponse.builder()
+                .sessionId(signUpSession.getSessionId())
+                .otpMediumValue(maskedIdentifier)
+                .otpMedium(otpMediumType.toString())
+                .expiryInMinutes(userService.getExpiryInMinutes())
+                .build());
+    }
+
+    private String maskedIdentifier(UserSignUpEnquiry userSignUpEnquiry) {
+        String maskedIdentifier = userSignUpEnquiry.getIdentifier();
+        if (userSignUpEnquiry.getIdentifierType().equals(IdentifierType.MOBILE.toString())) {
+            if (maskedIdentifier != null && !maskedIdentifier.equals("")) {
+                return maskedIdentifier.replaceAll("\\d(?=(?:\\D*\\d){4})", "*");
+            }
+        }
+        return null;
+    }
+
+    private OtpMediumType getOtpMedium(UserSignUpEnquiry userSignUpEnquiry) {
+        String identifierType = userSignUpEnquiry.getIdentifierType();
+        if (identifierType != null && identifierType.equals(IdentifierType.MOBILE.toString())) {
+            return OtpMediumType.MOBILE;
+        }
+        return null;
+    }
+
+    private Mono<UserSignUpEnquiry> getVerfiedIdentifier(Profile profile) {
+        Identifier verifiedIdentifier;
+        if (profile.getVerifiedIdentifiers() != null && !profile.getVerifiedIdentifiers().isEmpty()) {
+            List<Identifier> mobileIdentifiers = profile.getVerifiedIdentifiers().stream().filter(identifier ->
+                    identifier.getType() == IdentifierType.MOBILE).collect(Collectors.toList());
+            verifiedIdentifier = mobileIdentifiers.isEmpty() ?
+                    profile.getVerifiedIdentifiers().get(0) : mobileIdentifiers.get(0);
+
+            return Mono.just(UserSignUpEnquiry.builder()
+                    .identifier(verifiedIdentifier.getValue())
+                    .identifierType(verifiedIdentifier.getType().toString())
+                    .build());
+        }
+
+        return Mono.error(ClientError.unAuthorized());
+    }
+
+    @PutMapping("/profile/reset-password")
+    public Mono<Session> update(@RequestBody UpdateUserRequest request,
+                                @RequestHeader(name = "Authorization") String token) {
+        var updateUserRequests = UpdateUserRequestValidator.validatePassword(request.getPassword());
+        return updateUserRequests.isValid()
+                ? Mono.justOrEmpty(signupService.sessionFrom(token))
+                .flatMap(sessionId -> userService.update(request, sessionId)
+                        .zipWith(Mono.just(sessionId))
+                        .flatMap(tuple -> signupService.removeOf(tuple.getT2()).thenReturn(tuple.getT1())))
+                : Mono.error(new ClientError(BAD_REQUEST,
+                new ErrorRepresentation(new Error(INVALID_REQUESTER,
+                        updateUserRequests.getError()))));
```

Comment:
We can use `ClientError invalidRequester()` here. Or we can create a new static method for returning the client error in `ClientError.java`

Reply:
Error here is a combination of various scenarios for password validation. Message is computed dynamically based on the violated rules of password.

Reply:
Yes. We can overload the `ClientError.invalidRequester()` method to take the error as argument, right?

Reply:
done.